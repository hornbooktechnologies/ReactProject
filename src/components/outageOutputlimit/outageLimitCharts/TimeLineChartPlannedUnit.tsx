import React, { Dispatch, SetStateAction } from 'react'
import ReactApexChart from 'react-apexcharts'
import { useTranslation } from 'react-i18next'
import { TimeLineChartsList } from '../../../models/outageOutputLimit/TimelineChartModel'
import { DateformatConstant, ChartConstant, TIMEZONE } from '../../../utils/AppConstants'
import { getFiscalYearTimestamp } from '../../../utils/utils'

type Props = {
  className?: string
  timelinecharts: TimeLineChartsList
  selectYear: number
  setXaxisMin: Dispatch<SetStateAction<number>>
  setXaxisMax: Dispatch<SetStateAction<number>>
  xaxisMin: number
  xaxisMax: number
  isUnitWiseView?: boolean
}
type xasis = {
  type?: 'category' | 'datetime' | 'numeric' | undefined
}

const TimeLineChartPlannedUnit: React.FC<Props> = (props) => {
  const { i18n } = useTranslation();
  const yAxisLabelMinWidth = 90;
  const yAxisLabelMaxWidth = 130;
  const chartHeight = 180;
  const xaxisType: xasis = {
    type: 'datetime',
  }

  const annotation = {
    annotations: {
      xaxis: [
        {
          x: new Date().getTime() + TIMEZONE.JPN_TIMEZONE,
          strokeDashArray: 0,
          borderColor: ChartConstant.LABEL_COLOR,
          label: {
            borderColor: ChartConstant.LABEL_COLOR,
            style: {
              color: ChartConstant.LABEL_TEXT_COLOR,
              background: ChartConstant.LABEL_COLOR,
              fontFamily:  ChartConstant.ANNOTATION.FONT_FAMILY,
              fontSize: ChartConstant.ANNOTATION.FONT_SIZE,
            },
            text: i18n.t("LABELS.PRESENT"),
          }
        },
      ],
    },
  };

  let state = {
    options: {
      chart: {
        toolbar: {
          tools:{
            download: false,
          },
        },
        events: {
          beforeMount: function (chartContext: any, options: any) {
            props.setXaxisMin(options.config.xaxis.min)
            props.setXaxisMax(options.config.xaxis.max)
          },
          beforeZoom: function (chartContext: any, { xaxis }: any) {
            props.setXaxisMin(xaxis.min)
            props.setXaxisMax(xaxis.max)
          },
          beforeResetZoom: function () {
            props.setXaxisMin(getFiscalYearTimestamp(props.selectYear))
            props.setXaxisMax(getFiscalYearTimestamp(props.selectYear + 1))
            return {
              xaxis: {
                min: getFiscalYearTimestamp(props.selectYear),
                max: getFiscalYearTimestamp(props.selectYear + 1),
              },
            }
          },
          scrolled: function (chartContext: any, { xaxis }: any) {
            props.setXaxisMin(xaxis.min)
            props.setXaxisMax(xaxis.max)
          },
        },
      },
      plotOptions: {
        bar: {
          horizontal: true,
          rangeBarGroupRows: true,
        },
      },
      grid: {
        show: true,
        strokeDashArray: ChartConstant.TIMELINE_CHART_GRID_STROKE_AMOUNT,
        borderColor: ChartConstant.TICKLINE_STROKE,
      },
      legend: {
        show: false,
      },
      xaxis: {
        type: xaxisType.type,
        min: props.xaxisMin,
        max: props.xaxisMax,
        labels: {
          show: false,
        },
        axisBorder: {
          show: false,
        },
        axisTicks: {
          show: false,
        }
      },
      yaxis: {
        showForNullSeries: false,
        labels: {
          minWidth: yAxisLabelMinWidth,
          maxWidth: yAxisLabelMaxWidth,
          style: {
            colors: ChartConstant.WHITE_STROKE,
          },
        },
      },
      toolbar: {
        show: false,
      },
      tooltip: {
        x: {
          format: DateformatConstant.TIMELINE_CHART_TOOLTIP_FORMAT,
        },
      },
      fill: {
        opacity: ChartConstant.OPACITY,
      },
    },
    series: props.timelinecharts,
  }

  // showing annotation on timeline chart in case of unit wise view
  if (props.isUnitWiseView) {
    state.options = { ...state.options, ...annotation };
  }

  if (props.timelinecharts.length > 0) {
    return (
      <ReactApexChart
        className={['timeline-chart', props.className].join(' ')}
        options={state.options}
        series={state.series}
        type="rangeBar"
        height={chartHeight}
      />
    )
  } else {
    return null
  }
}

export default TimeLineChartPlannedUnit
