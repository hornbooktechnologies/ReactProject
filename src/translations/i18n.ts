import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
