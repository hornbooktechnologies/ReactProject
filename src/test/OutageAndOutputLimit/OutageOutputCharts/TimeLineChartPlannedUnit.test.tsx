import React from "react";
import renderer from "react-test-renderer";
import TimeLineChartPlannedUnit from "../../../components/outageOutputlimit/outageLimitCharts/TimeLineChartPlannedUnit";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: TimeLineChartPlannedUnit", () => {
  it("TimeLineChartPlannedUnit renders correctly", () => {
    const chartTestData = {
        timelinecharts: [],
        selectYear: 2021,
        xaxisMin: 2021,
        xaxisMax: 2022,
        setXaxisMin: () => {},
        setXaxisMax: () => {},
    }
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <TimeLineChartPlannedUnit {...chartTestData} />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
