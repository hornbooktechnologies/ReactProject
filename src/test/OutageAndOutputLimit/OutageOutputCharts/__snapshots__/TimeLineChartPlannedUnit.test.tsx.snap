// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TimeLineChartPlannedUnit TimeLineChartPlannedUnit renders correctly 1`] = `null`;
